etc/scripts/generate-github-test-data.py

```diff
@@ -0,0 +1,140 @@
+# HOW TO USE ME?
+# `python3 /Users/jkay/repos/github-for-jira/etc/scripts/generate-github-test-data.py
+# Generate a personal access token from GitHub(PAT)
+# replace ACCESS_TOKEN with the PAT
+# replace USERNAME with GitHub username
+#
+import requests
+import os
+import subprocess
+import time
+import random
+
+# GitHub API base URL
+BASE_URL = 'https://api.github.com'
+
+# GitHub access token
+ACCESS_TOKEN = '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'
+
+# GitHub username
+USERNAME = 'joshkay10'
+
+# Number of branches and loop iterations
+NUM_BRANCHES = 5
+LOOP_ITERATIONS = 3
+
+# Helper function to make authenticated API requests
+def make_api_request(method, url, headers=None, data=None):
+    headers = headers or {}
+    headers['Authorization'] = f'Bearer {ACCESS_TOKEN}'
+    response = requests.request(method, url, headers=headers, json=data)
+    try:
+        response.raise_for_status()
+        return response.json()
+    except requests.exceptions.HTTPError as e:
+        print(f'Error making API request: {e}')
+        print(f'Response: {response.text}')
+        raise
+
+# Helper function to execute bash commands
+def run_bash_command(command):
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    return process.returncode, stdout, stderr
+
+# Generate a unique repository name based on epoch time
+repo_name = f'repo-{int(time.time())}'
+
+try:
+    # Step 1: Create a new repository
+    repo_data = {
+        'name': repo_name
+    }
+    create_repo_url = f'{BASE_URL}/user/repos'
+    make_api_request('POST', create_repo_url, data=repo_data)
+    print(f'Repository "{repo_name}" created successfully.')
+
+    # Step 2: Initialize the repository, add README, and push initial commit
+    os.mkdir(repo_name)
+    os.chdir(repo_name)
+    os.system('git init')
+    os.system('echo "# README" >> README.md')
+    os.system('git add README.md')
+
+    print(f'Current working directory: {os.getcwd()}')
+    # Add main.yml to .github/workflows directory
+    workflow_file_name = 'main.yml'
+    workflow_file_path = os.path.join(os.getcwd(), '.github', 'workflows', workflow_file_name)
+    os.makedirs(os.path.dirname(workflow_file_path), exist_ok=True)
+
+    with open('../build-and-deploy-workflow-example.yml', 'r') as source_file:
+        content = source_file.read()
+        with open(workflow_file_path, 'w') as workflow_file:
+            workflow_file.write(content)
+
+    os.system(f'git add .github/workflows/{workflow_file_name}')
+    os.system('git commit -m "Add initial files"')
+    os.system(f'git branch -M main')
+    os.system(f'git remote add origin https://github.com/{USERNAME}/{repo_name}.git')
+    os.system(f'git push -u origin main')
+
+    for i in range(NUM_BRANCHES):
+        branch_name = f'branch-{i}'
+
+        # Step 3: Create a new branch
+        create_branch_command = f'git checkout -b {branch_name}'
```

Comment:
Should we consider converting some of the steps into functions to help with readability?

Reply:
agreed and updated, returned the usage comments up top too!

Reply:
Much more readable :hyperblob: lmk how you go with looking into those other libraries and whether you end up using them or not.